contracts/token/CRC1155/extensions/ICRC1155Metadata.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC1155/extensions/IERC1155MetadataURI.sol";
+
+/**
+ * @dev Optional metadata extension.
+ */
+interface ICRC1155Metadata is IERC1155MetadataURI {
+    /**
+     * @dev Returns the token collection name.
+     */
+    function name() external view returns (string memory);
+
+    /**
+     * @dev Returns the token collection symbol.
+     */
+    function symbol() external view returns (string memory);
```

Comment:
I guess the point here is that when CRC1155 is used as a CRC721 contract then it could have one symbol instead of separate symbols for each token type. Names and symbols for token types can be defined in the metadata, as specified in the ERC1155 standard.

Reply:
Exactly. By the way, ERC1155 standard do not have `name` and `symbol`, it only supports the `IERC1155MetadataURI` interface.

Reply:
The standard defines `name` for each `tokenId` in the metadata. You're right that it does not specify `symbol`.

```json
{
    "title": "Token Metadata",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "Identifies the asset to which this token represents"
        },
```

Reply:
This is similar to `IERC721Metadata`, the `name()` in interface is different from the `name` in JSON schema of token URI. The former one is the NFT token name, and the latter is name of specified token ID.